test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  compiler will attempt to implement its iterations using any
+  SIMD/SIMT parallelism that's available.  For example, if executing
+  the loop on a processor with vector instructions, it will attempt to
+  implement the loop using those instructions if possible.  Notably, a
+  foreach-loop will not implement its iterations using multiple Chapel
+  tasks or software threads (the forall-loop, below, does this).
+
+  Syntactically, foreach-loops are identical to for-loops, simply
+  using the ``foreach`` keyword.  For example, the following
+  ``foreach`` loop will double the values of the array ``A`` declared
+  above:
+
+*/
+
+  foreach a in A do
+    a *= 2;
+
+  writeln("After our first foreach-loop, A is: ", A);
+
+/*
+
+  Because each iteration is performing its own operations on its own
+  elements of A, this loop is trivially parallel-safe, and completely
+  reasonable to write using ``foreach``.  When running the computation
+  on a processor that supports vector instructions for performing
+  floating point multiplications, the ``foreach`` keyword's assertion
+  that the loop is legal to parallelize improves the compiler's
+  ability to implement the loop using those instructions.
+
+  Like the for-loops above, Chapel's foreach-loops support zippered
+  iteration.  For example, the following loop performs a zippered
+  iteration over the array `A` and an unbounded range `1..`.
+
+*/
+
+  foreach (a, i) in zip(A, 1..) do
+    a += (i / 100.0);
+
+   writeln("After our first zippered foreach-loop, A is: ", A);
+
+/*
+
+  Note that if a Chapel program only used foreach-loops to express its
+  parallelism, it would never make use of the multiple processor cores
+  of a modern processor nor the distinct compute nodes of a cluster or
+  HPC system.  This is because foreach-loops don't ever introduce new
+  Chapel tasks, and tasks are the only way to run in parallel at
+  system scales beyond a single processor.  As a result, to leverage
+  the full power of most parallel platforms, we need to look to
+  Chapel's other parallel loop forms.
+
+
+  forall-loops
+  ------------
+
+  Forall-loops are similar to foreach-loops, except that they have the
+  potential to be implemented using multiple Chapel tasks.  This
+  permits them to use multiple cores and/or compute nodes to execute
+  the loop's iterations.
+
+  Just as Chapel's for-loops invoke a serial iterator, its
+  forall-loops invoke a parallel iterator.  Where serial iterators may
+  only yield values sequentially, a parallel iterator's yield
+  statements may occur within parallel loops and constructs, resulting
+  in parallel execution.  When using a forall-loop with zippered
+  iterands, the first iterand in the zippering controls the loop's
+  parallelism.  For details about writing such parallel iterators, see
+  the :ref:`Parallel Iterators Primer <primers-parIters>`.
+
+  A parallel iterator can create as many tasks as it wants, and can
```

Comment:
I recommend talking about this after showing some simple forall loops that don't involve arrays.